packages/server/core/src/base/middlewares/renderHandler/render.ts

```diff
@@ -36,6 +38,35 @@ interface CreateRenderOptions {
   nonce?: string;
 }
 
+function getRouter(routes: ServerRoute[]): Router<ServerRoute> {
+  const sorted = routes.filter(r => !r.isApi).sort(sortRoutes);
+
+  const router = new TrieRouter<ServerRoute>();
+
+  for (const route of sorted) {
+    const { urlPath: originUrlPath } = route;
+
+    const urlPath = originUrlPath.endsWith('/')
+      ? `${originUrlPath}*`
+      : `${originUrlPath}/*`;
+    router.add('*', urlPath, route);
+  }
+
+  return router;
+}
+
+function matchRoute(
```

Comment:
In the public middleware, there is also a matchRoute, should this one be used as well?

Reply:
No, public middleware don't need handle dynamic params.
On the other hand, public middleware matchRoutes is not same as render middleware's